### PR TITLE
server/cluster: refine log errs in cluster

### PR DESCRIPTION
**Pull request by howardlau1999**
Signed-off-by: Howard Lau <howardlau1999@hotmail.com>

### What problem does this PR solve?

To solve #2704 

### Check List

Tests

- No test

### Release note

- No release note

### CODE REVIEW

File: server/cluster/cluster.go [hunk 1]
```diff
@@ -498,7 +499,7 @@ func (c *RaftCluster) HandleStoreHeartbeat(stats *pdpb.StoreStats) error {
 	}
 	if newStore.NeedPersist() && c.storage != nil {
 		if err := c.storage.SaveStore(store.GetMeta()); err != nil {
-			log.Error("failed to persist store", zap.Uint64("store-id", newStore.GetID()))
+			log.Error("failed to persist store", zap.Error(errs.ErrPersistStore.FastGenByArgs(newStore.GetID())))
```
Comment (ZenoTan) [inline server/cluster/cluster.go:502]: No `zap.NamedError("cause", err)` here = =

**Comment by howardlau1999** [comment]
@ZenoTan PTAL

**Comment by rleungx** [comment]
Please resolve the conflicts.

**Comment by howardlau1999** [comment]
@rleungx updated. PTAL

File: pkg/errs/errno.go [hunk 2]
```diff
@@ -65,7 +66,18 @@ var (
 
 // apiutil errors
 var (
-	ErrRequestHTTP   = errors.Normalize("send HTTP request failed", errors.RFCCodeText("PD:apiutil:ErrRequestHTTP"))
-	ErrReadHTTPBody  = errors.Normalize("read HTTP body failed", errors.RFCCodeText("PD:apiutil:ErrReadHTTPBody"))
-	ErrWriteHTTPBody = errors.Normalize("write HTTP body failed", errors.RFCCodeText("PD:apiutil:ErrWriteHTTPBody"))
+	ErrRequestHTTP    = errors.Normalize("send HTTP request failed", errors.RFCCodeText("PD:apiutil:ErrRequestHTTP"))
+	ErrReadHTTPBody   = errors.Normalize("read HTTP body failed", errors.RFCCodeText("PD:apiutil:ErrReadHTTPBody"))
+	ErrWriteHTTPBody  = errors.Normalize("write HTTP body failed", errors.RFCCodeText("PD:apiutil:ErrWriteHTTPBody"))
+	ErrNewHTTPRequest = errors.Normalize("new HTTP request failed", errors.RFCCodeText("PD:apiutil:ErrNewHTTPRequest"))
```
Comment (rleungx) [inline pkg/errs/errno.go:72]: should we put the error into this section?

Comment (howardlau1999) [reply pkg/errs/errno.go:72]: i think this error is related to HTTP requests and this section includes errors related to HTTP processing

Comment (rleungx) [reply pkg/errs/errno.go:72]: But `apiutil` is a package name, which may not be related to this error.

**Comment by rleungx** [comment]
CI failed. @howardlau1999

**Review by rleungx — APPROVED** [review]
LGTM

**Pull request merged**